### ISSUE TITLE
The "Usage Browser" shows 0 for all our "block list" aka element doctypes

### ISSUE THREAD

**Issue by kdx-perbol** [DanDiplo/Umbraco.GodMode#39]
Does the usage browser have blocklist doctype support? We have 0 usages for all blocktypes, i.e., element type doctypes that are used as blocks in blocklists.

Umbraco 11/12, latest GodMode.

**Comment by DanDiplo** [comment]
Hi. No, unfortunately it doesn't. This is due the way that blocks are currently stored in Umbraco. Normal content is stored in the Umbraco database - so there is essentially one row per item. But blocks are stored as "blobs" of JSON within those content pages. So there is no way of finding out where they are referenced. Neither the Umbraco API nor raw SQL can reveal this information, hence I can't display it.

There is a proposal for reusable content blocks that may come in Umbraco in the future - and in this they propose storing blocks the same ways content. This should then allow usage and references to be tracked, which will be great. But until then I don't think it's technically feasible. See https://github.com/umbraco/rfcs/blob/0025-reusable-content-with-global-blocks-2/cms/0025-reusable-content-with-global-blocks.md

**Comment by kdx-perbol** [comment]
> there is no way of finding [...] hence I can't display it.

OK. If this is true for all element doctypes, perhaps the GodMode UI could mention this or even exclude them from the usage browser?

**Comment by DanDiplo** [comment]
Fair point. I recall there is a notice on the datatype browser about this, as it can't "see" datatypes that are nested in block list etc.

If you use the "Type" filter at top and select "Content" then it will only show you actual document types, if that helps?

I'll leave this open and look into next time I work on it. Cheers.

**Comment by DanDiplo** [comment]
There's now a v17 that does a lot of this so I'm closing - https://marketplace.umbraco.com/package/diplo.godmode
